src/main/java/codeu/model/store/basic/ActivityStore.java

```diff
@@ -0,0 +1,146 @@
+    private static ActivityStore instance;
+
+    /**
+     * Returns the singleton instance of ActivityStore that should be shared between all servlet classes.
+     * Do not call this function from a test; use getTestInstance() instead.
+     */
+    public static ActivityStore getInstance() {
+        if (instance == null) {
+            instance = new ActivityStore(PersistentStorageAgent.getInstance());
+        }
+        return instance;
+    }
+
+    /**
+     * Instance getter function used for testing. Supply a mock for PersistentStorageAgent.
+     *
+     * @param persistentStorageAgent a mock used for testing
+     */
+    public static ActivityStore getTestInstance(PersistentStorageAgent persistentStorageAgent) {
+        return new ActivityStore(persistentStorageAgent);
+    }
+
+    /**
+     * The PersistentStorageAgent responsible for loading Users from and saving Users to Datastore.
+     */
+    private PersistentStorageAgent persistentStorageAgent;
+
+    /** The in-memory list of Activities. */
+    private List<Activity> activities;
+
+    /** This class is a singleton, so its constructor is private. Call getInstance() instead. */
+    private ActivityStore(PersistentStorageAgent persistentStorageAgent) {
+        this.persistentStorageAgent = persistentStorageAgent;
+        activities = new ArrayList<>();
+    }
+
+     /**
+     * Access all Activity objects.
+     */
+    public List<Activity> getAllActivities() {
+        return activities;
+    }
+
+    /**
+     * Access all Activity objects, sorted from oldest to most recent.
+     * Sorts all of the activities based on creation attribute.
+     */
+    public List<Activity> getAllSortedActivities() {
+        List<Activity> activitiesCopy = new ArrayList<>(activities);
+        activitiesCopy.sort(Collections.reverseOrder());
+        return activitiesCopy;
+    }
+
+    /**
+     * Access all Activity objects that were created on or after a specified date.
+     * 
+     * @param instant time that the Activity objects must be created on or after
+     */
+    public List<Activity> getActivitiesMadeOnOrAfter(Instant instant) {
+        int index = -1; // Index to take a sublist up to, exclusive
+        List<Activity> sortedActivities = getAllActivities();
+        for (int i=0; i < sortedActivities.size(); i++) {
+            if (sortedActivities.get(i).getCreationTime().compareTo(instant) < 0) {
+                index = i;
+                break;
+            }
+        }
+
+        if (index != -1) {
+            return sortedActivities.subList(0,index);
+        } else {
+            return sortedActivities;
+        }
+    }
+
+    /**
+     * Access a specified number of Activity objects from a specified index
+     * 
+     * @param startIndex index to begin subarray
+     * @param batchSize number of Activity objects to access
+     */
+    public List<Activity> getActivitiesBatch(int startIndex, int batchSize) {
```

Comment:
That's more of a future use method that may or may not be made use of. For example, right now we currently load all the activities, but if we had millions then you probably don't want to display all of them, so having the option to load the next x number of activities would use this method.

Reply:
Sounds good. Im interesting to see how this will be used. Like how will you decide to what "startIndex" is. And what happens if we add an activity that is an older timestamp than the previous activities[startIndex], in that case activities[startIndex] is the new value and activities[startIndex+1] is a new value. Does the developer need to know this and change their start index or their batchSize? Or maybe we just dont care at all about how the list changes at all?